### PR TITLE
🌱 Bump sigs.k8s.io/cluster-api to main@v1.8.0-rc.0.0.20240926091600-d495b4efe59f

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -123,7 +123,7 @@ GOTESTSUM_BIN := gotestsum
 GOTESTSUM := $(abspath $(TOOLS_BIN_DIR)/$(GOTESTSUM_BIN)-$(GOTESTSUM_VER))
 GOTESTSUM_PKG := gotest.tools/gotestsum
 
-CONVERSION_GEN_VER := v0.30.0
+CONVERSION_GEN_VER := v0.31.0
```
Comment (sbueringer) [inline Makefile:126]: Should we bump KUBEBUILDER_ENVTEST_KUBERNETES_VERSION too?

File: config/default/crd/bases/infrastructure.cluster.x-k8s.io_vsphereclusteridentities.yaml [hunk 2]
```diff
@@ -147,6 +146,7 @@ spec:
                         can be useful (see .node.status.conditions), the ability to deconflict is important.
                       type: string
                   required:
+                  - lastTransitionTime
```
Comment (sbueringer) [inline config/default/crd/bases/infrastructure.cluster.x-k8s.io_vsphereclusteridentities.yaml:149]: Let's adjust our copy of old conditions to avoid this (like core capi)

File: test/e2e/config/vsphere.yaml [hunk 3]
```diff
@@ -26,6 +26,15 @@ providers:
   - name: cluster-api
     type: CoreProvider
     versions:
+      - name: "v1.9.99"
+        value: "https://storage.googleapis.com/k8s-staging-cluster-api/components/main/core-components.yaml"
```
Comment (sbueringer) [inline test/e2e/config/vsphere.yaml:30]: I think it would be better to pin this to a nightly matching our go dependency to retain control over what we test + avoid that go and image get out of sync

File: test/e2e/config/vsphere.yaml [hunk 4]
```diff
@@ -26,6 +26,15 @@ providers:
   - name: cluster-api
     type: CoreProvider
     versions:
+      - name: "v1.9.99"
```
Comment (sbueringer) [inline test/e2e/config/vsphere.yaml:29]: Wondering if we have to make changes to our clusterctl upgrade tests (happy to pair on it later to figure it out)

Comment (chrischdi) [reply test/e2e/config/vsphere.yaml:29]: adjusted, it already is then always updating to v1.9.99 :-) so only adjusted the names of the tests to be correct

Comment (sbueringer) [reply test/e2e/config/vsphere.yaml:29]: Sounds good, will take another look (likely Monday though)

**Comment by sbueringer** [comment]
/test pull-cluster-api-provider-vsphere-test-main 

to get more data. Test results looked a bit strange for a flake though (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-vsphere/3206/pull-cluster-api-provider-vsphere-test-main/1839613278108520448)

**Comment by chrischdi** [comment]
Not a flake, expected change due to merged foreground deletion :-)

**Comment by sbueringer** [comment]
Definitely not a flake, but the unit test failures weren't expected because of foreground deletion right? https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-vsphere/3206/pull-cluster-api-provider-vsphere-test-main/1839618109850259456

File: controllers/controllers_suite_test.go [hunk 5]
```diff
@@ -92,7 +93,7 @@ func setup() {
 		panic(fmt.Sprintf("unable to setup ClusterCacheTracker: %v", err))
 	}
 
-	controllerOpts := controller.Options{MaxConcurrentReconciles: 10}
+	controllerOpts := controller.Options{MaxConcurrentReconciles: 10, SkipNameValidation: ptr.To(true)}
```
Comment (fabriziopandini) [inline controllers/controllers_suite_test.go:96]: Out of curiosity, why do we need this?

**Review by fabriziopandini — COMMENTED** [review]
Great work!

File: controllers/controllers_suite_test.go (hunk 5 above)
Comment (sbueringer) [reply controllers/controllers_suite_test.go:96]: controller-runtime added a new safeguard and it had to be done with a package-level singleton.

Basically if two controllers are build with the same name the second one will fail.

This was done to ensure that nobody builds & uses two controllers with the same name (to ensure unique metrics & logs).

Unfortunately that is a bit annoying in tests if controllers are build with the same name sequentially (package-level singleton detects that)

Comment (sbueringer) [reply controllers/controllers_suite_test.go:96]: xref: https://github.com/kubernetes-sigs/controller-runtime/pull/2902

**Comment by chrischdi** [comment]
Note: shadow CI jobs (e2e) are green too.

File: test/infrastructure/vcsim/config/crd/bases/vcsim.infrastructure.cluster.x-k8s.io_vcentersimulators.yaml [hunk 6]
```diff
@@ -86,7 +86,6 @@ spec:
                       For example: /DC0/host/DC0_C0/Resources/DC0_C0_RP1
                       Name prefix: RP, vcsim flag: -pool
                       Default: 0
-                      TODO: model pool selection for each cluster; for now ResourcePool named "Resources" will be always used
                         but ideally we should use RPx as per documentation above.
```
Comment (sbueringer) [inline test/infrastructure/vcsim/config/crd/bases/vcsim.infrastructure.cluster.x-k8s.io_vcentersimulators.yaml:89]: Same here, let's make this a single-line comment in go

File: test/infrastructure/vcsim/controllers/controlplaneendpoint_controller.go [hunk 7]
```diff
@@ -132,7 +132,7 @@ func (r *ControlPlaneEndpointReconciler) SetupWithManager(ctx context.Context, m
 	err := ctrl.NewControllerManagedBy(mgr).
 		For(&vcsimv1.ControlPlaneEndpoint{}).
 		WithOptions(options).
-		WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(ctrl.LoggerFrom(ctx), r.WatchFilterValue)).
+		WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(mgr.GetScheme(), ctrl.LoggerFrom(ctx), r.WatchFilterValue)).
```
Comment (sbueringer) [inline test/infrastructure/vcsim/controllers/controlplaneendpoint_controller.go:135]: I assume the change to pass in a different "predicateLog" is tracked somewhere and will be done in a separate PR?

(Just want to make sure we don't forget it)

**Comment by fabriziopandini** [comment]
lgtm pending @sbueringer comments

File: test/infrastructure/vcsim/controllers/controlplaneendpoint_controller.go (hunk 7 above)
Comment (chrischdi) [reply test/infrastructure/vcsim/controllers/controlplaneendpoint_controller.go]: Updated in this PR to set it everywhere.

File: test/infrastructure/net-operator/controllers/networkinterface_controller.go [hunk 8]
```diff
@@ -106,10 +106,12 @@ func (r *NetworkInterfaceReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 // SetupWithManager will add watches for this controller.
 func (r *NetworkInterfaceReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	predicateLog := ctrl.LoggerFrom(ctx).WithValues("controller", "net-operator")
```
Comment (sbueringer) [inline test/infrastructure/net-operator/controllers/networkinterface_controller.go:109]: The idea is for this k/v pair to match the logger that the reconciler gets, so I think this should be
```suggestion
	predicateLog := ctrl.LoggerFrom(ctx).WithValues("controller", "networkinterface")
```
(lower case kind of For, if Named doesn't overwrite it)

File: test/infrastructure/vcsim/controllers/vcsim_controller.go [hunk 9]
```diff
@@ -311,10 +311,12 @@ func (r *VCenterSimulatorReconciler) reconcileDelete(ctx context.Context, vCente
 
 // SetupWithManager will add watches for this controller.
 func (r *VCenterSimulatorReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	predicateLog := ctrl.LoggerFrom(ctx).WithValues("controller", "vcsim")
```
Comment (sbueringer) [inline test/infrastructure/vcsim/controllers/vcsim_controller.go:314]:
```suggestion
	predicateLog := ctrl.LoggerFrom(ctx).WithValues("controller", "vcentersimulator")
```

File: test/infrastructure/net-operator/controllers/networkinterface_controller.go (hunk 8 above)
Comment (sbueringer) [reply test/infrastructure/net-operator/controllers/networkinterface_controller.go:109]: If we don't match this exactly filtering on the controller name doesn't work (or we have to filter  on multiple controller names)

**Review by sbueringer — COMMENTED** [review]
Last nits

(we don't have to rerun all e2e tests after these)
